torch_xla/csrc/ir.h

```diff
@@ -60,16 +60,20 @@ using OutputMap =
     std::unordered_map<torch::lazy::Output, T, torch::lazy::Output::Hasher>;
 
 // Represents an input/operand for a Node object.
-struct Value {
+struct Value : public torch::lazy::Value {
   Value() = default;
-  Value(NodePtr node, size_t index = 0) : node(std::move(node)), index(index) {}
+  Value(NodePtr node, size_t index = 0)
+      : torch::lazy::Value(std::dynamic_pointer_cast<torch::lazy::Node>(node),
```

Comment:
The `dynamic_pointer_cast` here is needed because our `torch_xla::ir::NodePtr` is still pointing to `torch_xla::ir::Node`. 

I can follow-up with migrating to the upstream `torch::lazy::NodePtr` (was pushed off from the `Node` PR since it required many file changes) if we think it's necessary. Doing so will also help us to remove many of the functions in our `torch_xla:::ir::Value`.

Reply:
Agreed, sounds good. I'll merge this PR for now and follow-up with the `NodePtr`.